### DISCUSSION TITLE
Connecting Websockets client to Flask-sockeio server

### DISCUSSION THREAD

**Discussion by Ram-Krushana** [miguelgrinberg/Flask-SocketIO#1655] [Q&A]
I have created a demo application where I am trying to connect to the client of the WebSockets package. 

Client demo code.

        `import asyncio
        import websockets
        import logging
        logger = logging.getLogger('websockets')
        logger.setLevel(logging.INFO)
        logger.addHandler(logging.StreamHandler())
        async def hello():
            async with websockets.connect('ws://localhost:5002') as websocket:
                name = input("What's your name? ")
        
                await websocket.send(name)
                print(f"> {name}")
        
                greeting = await websocket.recv()
                print(f"< {greeting}")
      
      asyncio.get_event_loop().run_until_complete(hello())`

Server demo code.

      `from flask import Flask
      from flask_cors import CORS
      
      from flask_socketio import SocketIO
      
      app = Flask(__name__) 
      CORS(app)
      socketio = \
          SocketIO(
              app,
              logger=True,
              engineio_logger=True,
          )
      
      @app.route('/status', methods=['GET', 'POST'])
      def server_status():
          return "Server is running"
      
      @socketio.on('connect')
      def connect():
          print("Connect Message received:")
      if __name__ == '__main__':
          socketio.run(app, host="0.0.0.0", port=5002)`

There are no logs or error on the server-side. But I am getting connection close errors on client side.

        `Traceback (most recent call last):
          File "/home/pranavdave/work/projects/flask_sockeio_demo/client.py", line 52, in <module>
            asyncio.get_event_loop().run_until_complete(hello())
          File "/usr/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
            return future.result()
          File "/home/pranavdave/work/projects/flask_sockeio_demo/client.py", line 49, in hello
            greeting = await websocket.recv()
          File "/home/pranavdave/work/vEnv/py38/innovaccer/lib/python3.8/site-packages/websockets/legacy/protocol.py", line 421, in recv
            await self.ensure_open()
          File "/home/pranavdave/work/vEnv/py38/innovaccer/lib/python3.8/site-packages/websockets/legacy/protocol.py", line 726, in ensure_open
            raise self.connection_closed_exc()
        websockets.exceptions.ConnectionClosedError: code = 1005 (no status code [internal]), no reason
        
        Process finished with exit code 1`

I am using gevent in my flask-sockeio server.

Is my implementation is wrong or did I miss something to set up on server-side?

**Comment by miguelgrinberg** [comment]
Use a Socket.IO client. A WebSocket client cannot connect to a Socket.IO server. They're different protocols.

**Reply by Ram-Krushana** [reply]
Thank you for your quick reply.

**Reply by tsundokul** [reply]
thanks a lot, I've spent two days trying to figure out what's wrong because I've been using a simple WS client
